library/core/src/str/pattern.rs

```diff
@@ -1940,3 +1947,17 @@ unsafe fn small_slice_eq(x: &[u8], y: &[u8]) -> bool {
         vx == vy
     }
 }
+
+#[cfg(kani)]
+#[unstable(feature = "kani", issue = "none")]
+pub mod verify {
+    #[cfg(all(target_arch = "x86_64", target_feature = "sse2"))] // only called on x86
+    #[kani::proof]
+    pub fn check_small_slice_eq() {
```

Comment:
It might be nice to add a bit of non-determinism to the slice sizes. You can maybe use `any_slice`.

Reply:
Yes, it is a good idea. However, `any_slice_from_array` and `any_vec` are not in `kani_core`. So I copied their definitions from kani. Where is the better place to put them?

Reply:
Move them to `kani_core`. :smile:

Reply:
Yes, I created a PR for it. https://github.com/model-checking/kani/pull/3646